src/lib/cache_dir/dune

```diff
@@ -1,5 +1,5 @@
 (library
  (name cache_dir)
  (public_name cache_dir)
- (libraries core async coda_version logger)
- (preprocess (pps ppx_coda ppx_let ppx_custom_printf)))
+ (libraries core async coda_version logger blake2 consensus snark_params)
+ (preprocess (pps ppx_coda ppx_optcomp ppx_let ppx_custom_printf)))
```

Comment:
If you are going to `[%%import "/src/config.mlh"]` in a library, you have to also add a `(preprocessor_deps ...)` directive to the dune file of that library which points to the `config.mlh` file as well, otherwise switching dune profiles will corrupt the build.

(If you end up just using `Coda_compile_config` as I mentioned above, you can just skip this step since `Coda_compile_config` already depends on the `config.mlh` file).